src/Enterspeed.Source.SitecoreCms.V8/Services/EnterspeedConfigurationService.cs

```diff
@@ -0,0 +1,210 @@
+            //}
+            if (!IsConfigurationUpdated(enterspeedConfigurationItem, out DateTime _updatedDate))
+            {
+                return _configuration;
+            }
+
+            _configuration = new List<EnterspeedSitecoreConfiguration>();
+
+            foreach (Item enterspeedSiteConfigurationItem in enterspeedConfigurationItem.Children)
+            {
+                var config = new EnterspeedSitecoreConfiguration
+                {
+                    IsEnabled = true
+                };
+
+                config.IsPreview = enterspeedSiteConfigurationItem[EnterspeedIDs.Fields.EnterspeedEnablePreviewFieldID] == "1";
+
+                string configApiBaseUrl = enterspeedSiteConfigurationItem[EnterspeedIDs.Fields.EnterspeedApiBaseUrlFieldID];
+
+                config.BaseUrl = (configApiBaseUrl ?? string.Empty).Trim();
+
+                string configApiKey = enterspeedSiteConfigurationItem[EnterspeedIDs.Fields.EnterspeedApiKeyFieldID];
+                config.ApiKey = (configApiKey ?? string.Empty).Trim();
+
+                config.ItemNotFoundUrl = GetItemNotFoundUrl();
+
+                MultilistField enabledSitesField = enterspeedSiteConfigurationItem.Fields[EnterspeedIDs.Fields.EnterspeedEnabledSitesFieldID];
+
+                var enabledSites = enabledSitesField?.GetItems()?.ToList() ?? new List<Item>();
+                if (enabledSites.Any())
+                {
+                    List<SiteInfo> allSiteInfos = _siteContextFactory.GetSites();
+
+                    foreach (Item enabledSite in enabledSites)
+                    {
+                        var matchingSites = allSiteInfos.Where(x => x.RootPath.Equals(enabledSite.Paths.FullPath, StringComparison.OrdinalIgnoreCase));
+                        if (matchingSites.Count() == 0)
+                        {
+                            continue;
+                        }
+
+                        foreach (var matchingSite in matchingSites)
+                        {
+                            SiteContext siteContext = _siteContextFactory.GetSiteContext(matchingSite.Name);
+
+                            Language siteLanguage = _languageManager.GetLanguage(siteContext.Language);
+
+                            var languageEnterspeedSiteConfigurationItem = enterspeedSiteConfigurationItem.Database.GetItem(enterspeedSiteConfigurationItem.ID, siteLanguage);
+                            if (languageEnterspeedSiteConfigurationItem.Versions.Count == 0)
+                            {
+                                languageEnterspeedSiteConfigurationItem = enterspeedSiteConfigurationItem;
+                            }
+
+                            Item homeItem = _itemManager.GetItem(siteContext.StartPath, siteLanguage, Version.Latest, siteContext.Database);
+
+                            if (homeItem == null || homeItem.Versions.Count == 0)
+                            {
+                                // TODO - KEK: throw exception here?
```

Comment:
Do you think we should throw an exception here?

Reply:
@elglogins I have pushed the changes to the branch.